app/src/main/java/com/andryoga/safebox/data/db/secureDao/UserDetailsDaoSecure.kt

```diff
@@ -33,4 +36,16 @@ class UserDetailsDaoSecure @Inject constructor(
             )
         }
     }
+
+    private fun encrypt(userDetailsEntity: UserDetailsEntity): UserDetailsEntity {
+        userDetailsEntity.let {
+            return UserDetailsEntity(
+                it.key,
+                it.hint,
+                symmetricKeyUtils.encrypt(it.hint),
```

Comment:
is hint mandatory for user to enter?  if not then hint can be null also
see other secure dao to see how we handle encryption of optional fields